packages/autorest.python/autorest/codegen/models/base.py

```diff
@@ -187,3 +187,12 @@ def serialization_constraints(self) -> List[str]:
     @property
     def type_description(self) -> str:
         return self.type_annotation()
+
+    def contain_target(self, check_target: Callable[["BaseType"], bool]) -> bool:
```

Comment:
what is target?

This code is hard for me to read and understand. We shouldn't have private methods inherited across the different base classes. I feel like it's not necessary to have two methods, and a callable as an input. Shouldn't the model one just check if it's a ModelType, and you do something special in CombinedType?

Reply:
Hi @iscai-msft , I updated to remove private `_contain_target`.

For the `target`. I'm trying to create a flexible method to check whether a specified type of instance exists recursively, so I defined `contain_target` to accept a callback to validate whether a type instance is the target.

Currently there are two kinds of target need this logic: 1) ModelType, 2) named CombinedType. The usage codes are at https://github.com/Azure/autorest.python/blob/49c304c2b40b6614eed740116034080803489834/packages/autorest.python/autorest/codegen/models/property.py#L138-L148 .
We may need two different methods such as `contain_method` and `contain_named_union` in `BaseType` and derived types if we don't use this target?

Reply:
since these are checks to check what imports to add, shouldn't those imports be added in either `ModelType` or `CombinedType`?